main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"log"
+	"regexp"
+	"strconv"
+	"sync"
+)
+
+func Serve(w http.ResponseWriter, r *http.Request) {
+	var h http.Handler
+	p := r.URL.Path
+	switch {
+		case match(p, "/"):
+			h = get(Home)
+		case match(p, "/postings"):
+			h = get(GetLatestPostings)
+		default:
+			http.NotFound(w, r)
+			return
+	}
+	h.ServeHTTP(w, r)
+}
+
+// match reports whether path matches ^regex$, and if it matches,
+// assigns any capture groups to the *string or *int vars.
+func match(path, pattern string, vars ...interface{}) bool {
+	regex := mustCompileCached(pattern)
+	matches := regex.FindStringSubmatch(path)
+	if len(matches) <= 0 {
+		return false
+	}
+	for i, match := range matches[1:] {
+		switch p := vars[i].(type) {
+		case *string:
+			*p = match
+		case *int:
+			n, err := strconv.Atoi(match)
+			if err != nil {
+				return false
+			}
+			*p = n
+		default:
+			panic("vars must be *string or *int")
+		}
+	}
+	return true
+}
+
+var (
+	regexen = make(map[string]*regexp.Regexp)
+	relock  sync.Mutex
```

Comment:
The usual pattern here to make to ensure this is goroutine-safe is to:
- Embed the a sync.RWMutex (as opposed to a vanilla sync.Mutex) in the struct as a private (lowercase or) anonymous field.
- Embed the map as a private (lowercase or) anonymous field.
- Have get/set methods that call the lock for you. Because you use sync.RWMutex you can have multiple read locks that don't block xor a single writer. This way you only really lock when you're writing.

As you currently have it, you have a global mutable map that is not goroutine-safe outside of the function you intend to call it in. However, that doesn't stop other functions or methods down the line from breaking this leading to race conditions. That said, you are making it private to package so it's not as bad had this been public, but it's still best practice to manage mutability of your concurrent data structures even in trivial cases like this.